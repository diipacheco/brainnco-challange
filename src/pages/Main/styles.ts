import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;

  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: fixed;

  @media screen and (min-width: 700px) {
    flex-direction: row;
  }
`;

export const Content = styled.div`
  display: flex;

  width: 100%;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const DisclaimerMessage = styled.p`
  font-weight: 400;
  font-size: 11px;
  line-height: 21px;
  text-align: center;
  color: var(--black);
  margin-top: 50px;

  @media screen and (min-width: 700px) {
    position: absolute;
    top: 90%;
    margin: 0;
    font-size: 16px;
  }
`;
