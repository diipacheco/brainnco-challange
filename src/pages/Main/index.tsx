import React from 'react';

import ResultNumbers from '../../components/ResultNumbers';

import { Container, Content, DisclaimerMessage } from './styles';
import Header from '../../components/Header';

const Main: React.FC = () => {
  return (
    <Container>
      <Header />
      <Content>
        <ResultNumbers />
        <DisclaimerMessage>
          Este sorteio é meramente ilustrativo e não possui nenhuma ligação com
          a CAIXA.
        </DisclaimerMessage>
      </Content>
    </Container>
  );
};

export default Main;
