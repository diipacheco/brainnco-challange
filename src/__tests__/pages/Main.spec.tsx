import { render } from '@testing-library/react';

import Main from '../../pages/Main';

jest.mock('../../hooks/lotteries', () => ({
  useLotteries: () => ({
    lotteries: [
      {
        id: 0,
        nome: 'mega-sena',
      },
      {
        id: 1,
        nome: 'quina',
      },
      {
        id: 2,
        nome: 'lotofÃ¡cil',
      },
      {
        id: 3,
        nome: 'lotomania',
      },
      {
        id: 4,
        nome: 'timemania',
      },
      {
        id: 5,
        nome: 'dia de sorte',
      },
    ],
    fetchLotteries: jest.fn(),
    selectedLottery: {
      id: 0,
      nome: 'MEGA-SENA',
    },
    handleSelectedLottery: jest.fn(),
    fetchSelectedContests: jest.fn(),
    fetchContest: jest.fn(),
    selectedContest: {
      loteriaId: 0,
      concursoId: 2359,
    },
  }),
}));

describe('Main', () => {
  it('should match snapshots', () => {
    const wrapper = render(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
