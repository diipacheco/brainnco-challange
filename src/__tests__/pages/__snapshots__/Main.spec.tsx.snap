// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 13px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  list-style: none;
}

.c6 > li {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 8px 8px;
  width: 44px;
  height: 44px;
  background: var(--white);
  border-radius: 50%;
  font-weight: 700;
  font-size: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: fixed;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  font-weight: 400;
  font-size: 11px;
  line-height: 21px;
  text-align: center;
  color: var(--black);
  margin-top: 50px;
}

.c1 {
  width: 100%;
  height: 390px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background-color: var(--green-light);
}

.c2 {
  width: 100%;
  max-width: 215px;
  padding: 17px 33px;
  font-weight: 500;
  font-size: 15px;
  line-height: 18px;
  border-radius: 05px;
  background-color: var(--white);
  margin-top: 30px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 45px;
}

.c3 .logo-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 .logo-container > img {
  height: 55px;
}

.c3 .logo-container > h1 {
  color: var(--white);
  font-size: 26px;
  line-height: 37px;
  margin-top: 11px;
}

.c3 p {
  margin-top: 88px;
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: var(--white);
}

.c3 strong {
  display: block;
  font-size: 18px;
  line-height: 24px;
  color: #ffffff;
  margin-top: 06px;
}

@media screen and (min-width:700px) {
  .c5 {
    margin: 0;
  }
}

@media screen and (min-width:700px) {
  .c6 > li {
    width: 111px;
    height: 111px;
    margin: 10px 10px;
    font-size: 26px;
  }
}

@media screen and (min-width:700px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media screen and (min-width:700px) {
  .c7 {
    position: absolute;
    top: 90%;
    margin: 0;
    font-size: 16px;
  }
}

@media screen and (min-width:700px) {
  .c1 {
    height: 100vh;
    max-width: 550px;
  }

  .c1::after {
    content: '';
    width: 40px;
    height: 100%;
    background-color: var(--background);
    position: absolute;
    border-radius: 62%;
    right: -19px;
  }
}

@media screen and (max-width:700px) {
  .c1::after {
    content: '';
    width: 100%;
    height: 52px;
    background-color: var(--background);
    position: absolute;
    border-radius: 40%;
    top: 92%;
  }
}

@media screen and (min-width:700px) {
  .c3 {
    height: 100%;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <select
            class="c2"
            data-testid="select"
          >
            <option
              data-testid="selected-option-0"
              value="mega-sena"
            >
              MEGA-SENA
            </option>
            <option
              data-testid="selected-option-1"
              value="quina"
            >
              QUINA
            </option>
            <option
              data-testid="selected-option-2"
              value="lotofácil"
            >
              LOTOFÁCIL
            </option>
            <option
              data-testid="selected-option-3"
              value="lotomania"
            >
              LOTOMANIA
            </option>
            <option
              data-testid="selected-option-4"
              value="timemania"
            >
              TIMEMANIA
            </option>
            <option
              data-testid="selected-option-5"
              value="dia de sorte"
            >
              DIA DE SORTE
            </option>
          </select>
          <div
            class="c3"
          >
            <div
              class="logo-container"
            >
              <img
                alt="logo brain challenge"
                src="logo.svg"
              />
              <h1>
                MEGA-SENA
              </h1>
            </div>
            <div>
              <p>
                Concurso
              </p>
              <strong>
                2359 – undefined
              </strong>
            </div>
          </div>
        </div>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            <ul
              class="c6"
            />
          </div>
          <p
            class="c7"
          >
            Este sorteio é meramente ilustrativo e não possui nenhuma ligação com a CAIXA.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlnjwi cinSkD"
    >
      <div
        class="sc-jSFjdj dYmdWw"
      >
        <select
          class="sc-gKAaRy dWOBBH"
          data-testid="select"
        >
          <option
            data-testid="selected-option-0"
            value="mega-sena"
          >
            MEGA-SENA
          </option>
          <option
            data-testid="selected-option-1"
            value="quina"
          >
            QUINA
          </option>
          <option
            data-testid="selected-option-2"
            value="lotofácil"
          >
            LOTOFÁCIL
          </option>
          <option
            data-testid="selected-option-3"
            value="lotomania"
          >
            LOTOMANIA
          </option>
          <option
            data-testid="selected-option-4"
            value="timemania"
          >
            TIMEMANIA
          </option>
          <option
            data-testid="selected-option-5"
            value="dia de sorte"
          >
            DIA DE SORTE
          </option>
        </select>
        <div
          class="sc-iCoGMd bMZhjp"
        >
          <div
            class="logo-container"
          >
            <img
              alt="logo brain challenge"
              src="logo.svg"
            />
            <h1>
              MEGA-SENA
            </h1>
          </div>
          <div>
            <p>
              Concurso
            </p>
            <strong>
              2359 – undefined
            </strong>
          </div>
        </div>
      </div>
      <div
        class="sc-hKFxyN bSOnKf"
      >
        <div
          class="sc-bdnxRM fkYXbU"
        >
          <ul
            class="sc-gtsrHT juOHwc"
          />
        </div>
        <p
          class="sc-eCApnc hYRXVi"
        >
          Este sorteio é meramente ilustrativo e não possui nenhuma ligação com a CAIXA.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
